lib/async/redis/protocol/resp.rb

```diff
@@ -137,6 +140,19 @@ def write_lines(*args)
 						end
 					end
 				end
+
+				# This method is useful for pipelining because instead of flushing once
+				# per command, we want to flush just once for the whole pipeline for
+				# performance reasons.
+				def write_request_without_flush(arguments)
```

Comment:
I changed this. I added an optional `flush` arg that defaults to `true` as you suggested.
I left the `write_pipeline` in the `RESP` class. I think it's a bit better this way because then callers do not need to remember about calling `flush`.  But I'm open to suggestions, can you take a look at the changes and let me know what you think?